### DISCUSSION TITLE
accessing environment variables from shebang recipes

### DISCUSSION THREAD

**Discussion by willie** [casey/just#1989] [Q&A] [answered]
I have a shebang recipe that looks like this:

```
#!/usr/bin/env sh
set -euxo pipefail
berglas exec -- env | grep AUTH_
```

where berglas is a tool that decodes env vars with specific urls. Output from this is:

```
AUTH_USER=XXXX
AUTH_PASSWORD=XXXX
```

which is what I expect. However, I need to use the AUTH_USER and AUTH_PASSWORD in a call to curl. This is what I am trying:

```
#!/usr/bin/env sh
set -euxo pipefail
berglas exec -- curl -v -u ${AUTH_USER}:${AUTH_PASSWORD} some_url
```

and what I get is:

```
/var/folders/j7/00b7qgfn7szgn9jj1h_9bfmh0000gn/T/just-rgXt8Y/curl: line 46: AUTH_USER: unbound variable
```

How do I tell just to let the shell do the substitution for that env var reference?

**Comment by willie** [comment]
Another example:

```
#!/usr/bin/env sh
set -eux
berglas exec -- curl -v -u $AUTH_USER:$AUTH_PASSWORD some_url
```

errors with

```
/var/folders/j7/00b7qgfn7szgn9jj1h_9bfmh0000gn/T/just-ypH6pc/curl: line 48: AUTH_USER: unbound variable
```

**Comment by laniakea64** [comment, marked as answer]
If this is all one recipe, you could do something like
```
#!/usr/bin/env sh
set -euxo pipefail
export $(berglas exec -- env | grep AUTH_)
curl -v -u ${AUTH_USER}:${AUTH_PASSWORD} some_url
```
This is just the basic idea.  That `export` line might be unsound if any of the `AUTH_*` variables could contain characters that are special to the shell, but there are other ways to extract the values from that command output for passing to `export`.

**Reply by willie** [reply]
That is a super clever solution. It's solves the problem, but not in the way I had expected. Do you know of a way to prevent echo of the output to the terminal from the result of `env | grep AUTH_`?

**Reply by laniakea64** [reply]
> Do you know of a way to prevent echo of the output to the terminal from the result of `env | grep AUTH_`?

Try removing `x` from the `set` line?

**Reply by willie** [reply]
Perfect, thank you!
